lib/routes/account.js

```diff
@@ -1745,18 +1781,68 @@ module.exports = (
 
         db.account(uid)
           .then(
-            (account) => {
+            (accountResult) => {
+              account = accountResult
+
+              isAccountVerification = butil.buffersAreEqual(code, account.emailCode)
+
               // This endpoint is not authenticated, so we need to look up
               // the target email address before we can check it with customs.
               return customs.check(request, account.email, 'recoveryEmailVerifyCode')
-                .then(
-                  () => account
-                )
             }
           )
+          .then(() => {
+
+            if (! isAccountVerification) {
```

Comment:
Hm seems to be a fork in decision making here. Is this basically `IF isAccountVerification do X`, `IF isEmailVerification do Y`? 

can this be refactored?

Current code seems to be:

```
Request -> Account -> If isEmailVerification do X -> if isAccountVerification do Y -> Reply
```

Can we refactor into clean functions so it looks more like:
```
Request -> Account -> If isEmailVerification do X   ->           Reply
                                  -> if isAccountVerification do Y --------- ^
```

It is especially confusion because both branches call `db.verifyEmail(account, code)`, does it mean both of them will never be called together?